### DISCUSSION TITLE
Dedicated server

### DISCUSSION THREAD

**Discussion by codarth** [metaworking/channeld-ue-plugin#106] [Q&A]
From what I have seen so far with the plugin with UE5 (tried on 5.3), things look promising for our project. I have been able to get things running locally as explained in the getting started, however, before we utilize it for our project, we need to know how to get it running in a dedicated setting. I could be wrong but the only information I can find about this is in the Plugin Introduction | Cloud Deployment Tool. I am not sure if I am even packaging things right.

I have been able to package the project under Linus Server target and have been trying to use docker build. Perhaps I am missing something or I don't know docker like I should, but I can not get it to work on docker. 

Can I get some clarification on this : "then use docker build to build channeld and the game server images." Do I need to build channeld separate from the game server? I my head I need three builds, channeld, server, client. 

"Channeld Remote Image Tag and Server Remote Image Tag" are these settings in Unreal? If so I can not find them.

My goal was to have it running on a dedicated server (locally if I can), see if I can connect and see if others can as well. If we can do this then we may proceed with utilizing it in a project.

In conclusion, can I get instructions on how I can package, build and deploy to a dedicated server and connect with a clients (local and remote)?

**Comment by indiest** [comment]
Looks like your are looking for a way to deploy the servers (channeld + the dedicated servers) in a LAN environment. I would recommend using [docker compose](https://docs.docker.com/compose/) to achieve this, and [ChanneldUE Demo Project](https://github.com/channeldorg/channeld-ue-demos/tree/release/Packages) comes alone with the YAML file used by docker compose. You can copy the `Packages` folder into your own project, and modify `docker-compose.yml` to replace the UE project and map names with your own.

After you package the project under Linux Server target into the `Packages` folder, you should be able to deploy the servers using `docker compose up -d tps-server`.

To connect a client to the deployment, you can either set `Channeld Ip for Client` with the LAN IP of the server machine in the Project Settings and launch a PIE client, or run a packaged client with `channeldClientIp=<SERVER_IP>` argument.
